src/server/launchpad/resources.js

```diff
@@ -9,11 +9,21 @@ import { normalizeURI } from './uri';
 /** The base class for objects retrieved from Launchpad's web service. */
 export class Resource {
   /** Initialize a resource with its representation and URI. */
-  init(client, representation, uri) {
-    this.lp_client = client;
-    this.uri = uri;
-    for (const key of Object.keys(representation)) {
-      this[key] = representation[key];
+  init(client, uri, representation) {
+    // defining internal properties as non-enumerable not to leak any secrets
+    Object.defineProperty(this, 'lp_client', {
+      value: client,
+      enumerable: false
+    });
+    Object.defineProperty(this, 'uri', {
+      value: uri,
+      enumerable: false
+    });
+
+    if (representation) {
```

Comment:
Making `representation` optional, so that collection `Entry` can call it to get `lp_client` and `uri` without setting `representation`.